src/openlrr/game/interface/Priorities.cpp

```diff
@@ -28,45 +39,361 @@ LegoRR::Priorities_Globs & LegoRR::prioritiesGlobs = *(LegoRR::Priorities_Globs*
+				SFX_GetType(stringPartsSub[1], &prioritiesGlobs.sfxPriorityName[priorityTypeLoc]);;
+			}
+
+			prioritiesGlobs.priorityImage[priorityType] =      Gods98::Image_LoadBMPScaled(stringParts[1], 0, 0);
+			prioritiesGlobs.priorityPressImage[priorityType] = Gods98::Image_LoadBMPScaled(stringParts[2], 0, 0);
+			prioritiesGlobs.priorityOffImage[priorityType] =   Gods98::Image_LoadBMPScaled(stringParts[3], 0, 0);
+		}
+	}
+}
 
 // <LegoRR.exe @0045d1c0>
-//void __cdecl LegoRR::Priorities_Reset(void);
+void __cdecl LegoRR::Priorities_Reset(void)
+{
+	for (uint32 i = 0; i < prioritiesGlobs.count; i++) {
+		prioritiesGlobs.buttonTypes[i] = prioritiesGlobs.initialTypes[i];
+		AITask_Game_SetPriorityValue(prioritiesGlobs.buttonTypes[i], prioritiesGlobs.initialValues[i]);
+		AITask_Game_SetPriorityOff(prioritiesGlobs.buttonTypes[i], prioritiesGlobs.initialOff[i]);
+	}
+}
 
 // <LegoRR.exe @0045d210>
-//bool32 __cdecl LegoRR::Priorities_LoadLevel(const Gods98::Config* config, const char* gameName, const char* levelName);
+bool32 __cdecl LegoRR::Priorities_LoadLevel(const Gods98::Config* config, const char* gameName, const char* levelName)
+{
+	char* arrayID;
+	const Gods98::Config* prop;
+	prioritiesGlobs.count = 0;
+	arrayID = (char*)Config_ID(gameName, levelName, "Priorities");
+	// -FC, create this instead of reusing levelName pointer 10 times
+	AI_Priority locAiPro;
+	for (prop = Config_FindArray(config, arrayID); prop != nullptr; prop = Config_GetNextItem(prop)) {
+		if (AIPriority_GetType(Gods98::Config_GetItemName(prop), &locAiPro)) {
+			prioritiesGlobs.buttonTypes[prioritiesGlobs.count] = locAiPro;
+			prioritiesGlobs.initialTypes[prioritiesGlobs.count] = locAiPro;
+			// -FC, prios start at 80 and dec by 5 each
+			prioritiesGlobs.initialValues[prioritiesGlobs.count] =	(prioritiesGlobs.count * -PRIO_INC) + MAX_PRIO;
+			AITask_Game_SetPriorityValue(locAiPro,prioritiesGlobs.initialValues[prioritiesGlobs.count]);
+			if (Gods98::Util_GetBoolFromString(Gods98::Config_GetDataString(prop)) == BOOL3_TRUE) {
+				/* Priority needs to be explicitly defined as ON in Lego.cfg */
+				AITask_Game_SetPriorityOff(locAiPro, FALSE);
+				prioritiesGlobs.initialOff[prioritiesGlobs.count] = FALSE;
+			}
+			else {
+				AITask_Game_SetPriorityOff(locAiPro, TRUE);
+				prioritiesGlobs.initialOff[prioritiesGlobs.count] = TRUE;
+			}
+			prioritiesGlobs.count++;
+		}
+	}
+	return TRUE;
+}
 
 // <LegoRR.exe @0045d320>
-//void __cdecl LegoRR::Priorities_LoadPositions(const Gods98::Config* config, const char* gameName);
+void __cdecl LegoRR::Priorities_LoadPositions(const Gods98::Config* config, const char* gameName)
+{
+	char* keyPath;
+	char* parts[16];
+	char* startGameName = (char*)gameName;
+
+	for (uint32 i = 0; i < AI_Priority_Count; i++) {
+		char orderName[12]; // Max number of characters +1 that can appear in signed int32.
+		_itoa((int)i+1, orderName, 10);
+		keyPath = (char*)Config_ID(startGameName, "PrioritiesImagePositions", &orderName);
```

Comment:
`keyPath` needs to be const and `(char*)` cast should be removed.

`startGameName` isn't needed anymore, so pass `gameName` instead and remove the local variable.